pkg/ocm/servicelog.go

```diff
@@ -0,0 +1,88 @@
+package ocm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/ocm-agent-operator/api/v1alpha1"
+	"github.com/openshift/ocm-agent/pkg/consts"
+	"github.com/prometheus/alertmanager/template"
+)
+
+const (
+	ServiceLogActivePrefix  = "Issue Notification"
+	ServiceLogResolvePrefix = "Issue Resolution"
+)
+
+type ServiceLog struct {
+	ServiceName   string                               `json:"service_name"`
+	ClusterUUID   string                               `json:"cluster_uuid,omitempty"`
+	Summary       string                               `json:"summary"`
+	Description   string                               `json:"description"`
+	InternalOnly  bool                                 `json:"internal_only"`
+	Severity      v1alpha1.NotificationSeverity        `json:"severity"`
+	LogType       string                               `json:"log_type,omitempty"`
+	DocReferences []v1alpha1.NotificationReferenceType `json:"doc_references,omitempty"`
+}
+
+func NewServiceLog(summary, firingDesc, resolveDesc, clusterID string, severity v1alpha1.NotificationSeverity, logType string, references []v1alpha1.NotificationReferenceType, firing bool) *ServiceLog {
+	sl := ServiceLog{
+		ServiceName:   consts.ServiceLogServiceName,
+		ClusterUUID:   clusterID,
+		Summary:       summary,
+		InternalOnly:  false,
+		Severity:      severity,
+		LogType:       logType,
+		DocReferences: references,
+	}
+
+	// Use different Summary and Description for firing and resolved status for an alert
+	if firing {
+		sl.Description = firingDesc
+		sl.Summary = ServiceLogActivePrefix + ": " + summary
+	} else {
+		sl.Description = resolveDesc
+		sl.Summary = ServiceLogResolvePrefix + ": " + summary
+	}
+
+	return &sl
+}
+
+var (
+	slVarRefRe = regexp.MustCompile(`\${[^{}]*}`)
+)
+
+func (sl *ServiceLog) ReplacePlaceHoldersWithAlertLabelsAndAnnotations(alert *template.Alert) error {
+	replacePlaceHoldersInString := func(s string) (string, error) {
+		var err error
+		resolvePlaceHolder := func(placeHolder string) string {
+			if err == nil {
```

Comment:
What's the  `err == nil` meaning here? When is it not nil, if I'm not mistaken, it seems to just take the nil declaration from above?